wifi.go

```diff
@@ -0,0 +1,134 @@
+package keylight
+
+import (
+	"bytes"
+	"context"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+// A WiFiInfo contains the information neccessary for a Key Light to connect
+// to a wireless network.
+type WiFiInfo struct {
+	SSID         string `json:"ssid,omitempty"`
+	Passphrase   string `json:"passphrase,omitempty"`
+	SecurityType int    `json:"securityType,omitempty"`
+}
+
+const (
+	SecurityTypeNone int = 0
+	SecurityTypeWEP  int = 1
+	SecurityTypePSK  int = 2
+)
+
+// SetWiFiInfo encrypts and updates a Key Light's WiFi configuration.
+func (c *Client) SetWiFiInfo(ctx context.Context, wifi *WiFiInfo, device *Device) error {
+	b, err := json.Marshal(wifi)
+	if err != nil {
+		return err
+	}
+
+	blen := len(b)
+	padlen := aes.BlockSize - (blen % aes.BlockSize)
+	pad := bytes.Repeat([]byte{0}, padlen)
+
+	plaintext := append(b, pad...)
+	key := aesKey(device.HardwareBoardType, device.FirmwareBuildNumber)
+
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return err
+	}
+
+	ciphertext := make([]byte, aes.BlockSize+len(plaintext))
+	iv := ciphertext[:aes.BlockSize]
+	if _, err := io.ReadFull(rand.Reader, iv); err != nil {
+		return err
+	}
+
+	mode := cipher.NewCBCEncrypter(block, iv)
+	mode.CryptBlocks(ciphertext[aes.BlockSize:], plaintext)
+
+	return c.do(ctx, http.MethodPost, "/elgato/wifi-info", bytes.NewReader(ciphertext), nil)
+}
+
+// GetWiFiInfo fetches and decrypts a Key Light's WiFi configuration.
+func (c *Client) GetWiFiInfo(ctx context.Context, device *Device) (*WiFiInfo, error) {
+	ciphertext, err := c.doRaw(ctx, http.MethodGet, "/elgato/wifi-info", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	key := aesKey(device.HardwareBoardType, device.FirmwareBuildNumber)
+
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(ciphertext) < aes.BlockSize {
+		return nil, fmt.Errorf("ciphertext too short")
+	}
+	iv := ciphertext[:aes.BlockSize]
+	ciphertext = ciphertext[aes.BlockSize:]
+
+	if len(ciphertext)%aes.BlockSize != 0 {
+		return nil, fmt.Errorf("ciphertext is not a multiple of the block size")
```

Comment:
I have been prefixing errors with `keylight:` and let's switch to errors.New if there's no formatting need.

Reply:
Include something about "WiFi info".